spectral_cube/spectral_cube.py

```diff
@@ -2292,7 +2279,17 @@ def __init__(self, data, wcs, mask=None, meta=None, fill_value=np.nan,
                                            **kwargs)
 
         # Beam loading must happen *after* WCS is read
-        self.attach_beam(beam=beam)
+
+        if beam is None:
+            beam = cube_utils.try_load_beam(self.header)
+        else:
+            if not isinstance(beam, Beam):
+                raise TypeError("beam must be a radio_beam.Beam object.")
+
+        if beam is not None:
```

Comment:
we're already in the `else` of `if beam is None`, so this `if` could just be deleted with no change of indentation, I think

Reply:
no, I'll have to duplicate the three lines below that `if` without it for when the beam is loaded from the header. The other option would be to move loading a beam from the header somewhere into `io.fits`

Reply:
right, good point, OK.